dot/sync/chain_processor.go

```diff
@@ -42,24 +44,35 @@ type chainProcessor struct {
 	telemetry          telemetry.Client
 }
 
-func newChainProcessor(readyBlocks *blockQueue, pendingBlocks DisjointBlockSet,
-	blockState BlockState, storageState StorageState,
-	transactionState TransactionState, babeVerifier BabeVerifier,
-	finalityGadget FinalityGadget, blockImportHandler BlockImportHandler, telemetry telemetry.Client) *chainProcessor {
+type chainProcessorConfig struct {
+	readyBlocks        *blockQueue
+	pendingBlocks      DisjointBlockSet
+	syncer             ChainSync
+	blockState         BlockState
+	storageState       StorageState
+	transactionState   TransactionState
+	babeVerifier       BabeVerifier
+	finalityGadget     FinalityGadget
+	blockImportHandler BlockImportHandler
+	telemetry          telemetry.Client
+}
+
+func newChainProcessor(cfg *chainProcessorConfig) *chainProcessor {
```

Comment:
nit I'd revert this, I don't think this delta is needed nor related to this PR. Especially for unexported constructors, I'd say it's fine to list many arguments.

Reply:
I will keep it as it is because we already use a configuration struct for the `chain_sync` so adding this configuration struct for `chain_processor` keep them in the same pattern. I have changed to accept a value instead of a pointer.

Reply:
Ok, although it's still kinda out of scope. Would you mind opening another (easy) PR for those config changes?

Reply:
For PR size you're right, it's more for git history/git blame in IDE. It happens from time to time someone checks some old commit for X or Y reason, and it helps when it's focused. It's also fine to do side code changes if you need to change that code too, for example refactor an affected test